cypress/e2e/po/extensions/kubewarden/kubewarden.utils.ts

```diff
@@ -0,0 +1,245 @@
+import PagePo from '@/cypress/e2e/po/pages/page.po';
+import NameNsDescription from '@/cypress/e2e/po/components/name-ns-description.po';
+import BaseResourceList from '@/cypress/e2e/po/lists/base-resource-list.po';
+import CodeMirrorPo from '@/cypress/e2e/po/components/code-mirror.po';
+
+class KubewardenDashboardPagePo extends PagePo {
+  static url = '/c/local/kubewarden';
+  static goTo(): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(KubewardenDashboardPagePo.url);
+  }
+
+  constructor() {
+    super(KubewardenDashboardPagePo.url);
+  }
+
+  waitForTitlePreControllerInstall(): Cypress.Chainable {
+    return this.self().find('[data-testid="kw-install-title"]').should('contain', 'Kubewarden');
+  }
+
+  waitForTitleAfterControllerInstall(): Cypress.Chainable {
+    return this.self().find('[data-testid="kw-dashboard-title"]').should('contain', 'Welcome to Kubewarden');
+  }
+
+  startBackendInstallClick(): Cypress.Chainable {
+    return this.self().getId('kw-initial-install-button').click();
+  }
+
+  openTerminalClick() {
+    return this.self().getId('kw-cm-open-shell').click();
+  }
+
+  addKwRepoClick() {
+    return this.self().getId('kw-repo-add-button').click();
+  }
+
+  installOperatorBtnClick(): Cypress.Chainable {
+    return this.self().getId('kw-app-install-button').click();
+  }
+
+  defaultPolicyServerInstallClick() {
+    return this.self().getId('kw-defaults-banner-button').click();
+  }
+}
+
+class KubewardenPolicyServerListPagePo extends PagePo {
+  static url = '/c/local/kubewarden/policies.kubewarden.io.policyserver';
+  static goTo(): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(KubewardenPolicyServerListPagePo.url);
+  }
+
+  constructor() {
+    super(KubewardenPolicyServerListPagePo.url);
+  }
+}
+
+class KubewardenPolicyServerEditPagePo extends PagePo {
+  private static createPath(clusterId: string, id?: string ) {
+    const root = `/c/${ clusterId }/kubewarden/policies.kubewarden.io.policyserver`;
+
+    return id ? `${ root }/${ id }?mode=edit` : `${ root }/create`;
+  }
+
+  goTo(clusterId: string, id?:string): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(KubewardenPolicyServerEditPagePo.createPath(clusterId, id));
+  }
+
+  constructor(clusterId = 'local', id?: string) {
+    super(KubewardenPolicyServerEditPagePo.createPath(clusterId, id));
+  }
+
+  editYamlBtnClick():Cypress.Chainable {
+    return this.self().get('[data-testid="kw-policy-server-config-yaml-option"] [data-testid="button-group-child-1"]').click();
+  }
+
+  // the edit view of the policy server is not a generic resource edit/yaml view, so let's go
+  // straight to the source
+  codeMirror() {
+    return CodeMirrorPo.bySelector(this.self(), '[data-testid="yaml-editor-code-mirror"]');
+  }
+
+  saveForm(): AsyncButtonPo {
+    return new AsyncButtonPo('[data-testid="form-save"]', this.self());
+  }
+
+  waitForPolicyServerCreation(interceptName: string, nameToCheck: string) {
```

Comment:
cruresource has an alt push save button function that also waits for a http request saveAndWaitForRequests

Reply:
True... But this goes a bit farther deep and has custom checks on the expect. It's not as general as the `saveAndWaitForRequests` though.